rust/otap-dataflow/crates/signal-type-router/src/processor.rs

```diff
@@ -0,0 +1,215 @@
+            .send_message(data)
+            .await
+            .map_err(|e| SignalTypeRouterError::Engine {
+                source: Box::new(e),
+            })?;
+
+        Ok(())
+    }
+}
+
+/// Trait to abstract over different effect handler types
+trait EffectHandlerTrait {
+    async fn send_message(&mut self, data: OtapPdata) -> Result<(), EngineError<OtapPdata>>;
+}
+
+impl EffectHandlerTrait for LocalEffectHandler<OtapPdata> {
+    async fn send_message(&mut self, data: OtapPdata) -> Result<(), EngineError<OtapPdata>> {
+        LocalEffectHandler::send_message(self, data).await
+    }
+}
+
+impl EffectHandlerTrait for SharedEffectHandler<OtapPdata> {
+    async fn send_message(&mut self, data: OtapPdata) -> Result<(), EngineError<OtapPdata>> {
+        SharedEffectHandler::send_message(self, data)
+            .await
+            .map_err(EngineError::ChannelSendError)
+    }
+}
+
+#[async_trait(?Send)]
+impl LocalProcessor<OtapPdata> for SignalTypeRouter {
+    async fn process(
+        &mut self,
+        msg: Message<OtapPdata>,
+        effect_handler: &mut LocalEffectHandler<OtapPdata>,
+    ) -> Result<(), EngineError<OtapPdata>> {
+        match msg {
+            Message::Control(ctrl_msg) => {
+                log::debug!("SignalTypeRouter received control message: {ctrl_msg:?}");
+                // Handle control messages (shutdown, config updates, etc.)
+                // For now, we don't need to process control messages specifically
+                Ok(())
+            }
+            Message::PData(data) => {
+                log::trace!("SignalTypeRouter processing signal data");
+                self.route_signal(data, effect_handler).await.map_err(|e| {
+                    EngineError::ProcessorError {
+                        processor: "SignalTypeRouter".into(),
+                        error: e.to_string(),
+                    }
+                })
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl SharedProcessor<OtapPdata> for SignalTypeRouter {
+    async fn process(
+        &mut self,
+        msg: Message<OtapPdata>,
+        effect_handler: &mut SharedEffectHandler<OtapPdata>,
+    ) -> Result<(), EngineError<OtapPdata>> {
+        match msg {
+            Message::Control(ctrl_msg) => {
+                log::debug!("SignalTypeRouter received control message: {ctrl_msg:?}");
+                // Handle control messages (shutdown, config updates, etc.)
+                // For now, we don't need to process control messages specifically
+                Ok(())
+            }
+            Message::PData(data) => {
+                log::trace!("SignalTypeRouter processing signal data");
+                self.route_signal(data, effect_handler).await.map_err(|e| {
+                    EngineError::ProcessorError {
+                        processor: "SignalTypeRouter".into(),
+                        error: e.to_string(),
+                    }
+                })
+            }
+        }
+    }
+}
```

Comment:
We don't need to implement a Local and a Shared processor implementations. If you can implement a local processor then that's the only one to implement. If your processor requires a `Send` context then you implements a Shared processor. 
Please remove this code.

Reply:
Done. Local-only implementation; Shared impl removed/not included.